tests/foreman/cli/test_satellitesync.py

```diff
@@ -1139,6 +1139,92 @@ def test_positive_export_import_redhat_cv_with_huge_contents(
         for item in ['packages', 'source-rpms', 'package-groups', 'errata', 'module-streams']:
             assert exported_repo['content-counts'][item] == imported_repo['content-counts'][item]
 
+    @pytest.mark.tier2
+    def test_positive_export_cv_with_on_demand_repo(
+        self, export_import_cleanup_function, target_sat, module_org
+    ):
+        """Exporting CV version skips on_demand repo
+
+        :id: c366ace5-1fde-4ae7-9e84-afe58c06c0ca
+
+        :steps:
+
+            1. Create product
+            2. Create repos with immediate and on_demand download policy
+            3. Sync the repositories
+            4. Create CV with above product and publish
+            5. Attempt to export CV version contents
+
+        :expectedresults:
+
+            1. Export passes with warning and skips the on_demand repo
+
+        :BZ: 1998626
+        """
+        # Create custom product
+        product = make_product({'organization-id': module_org.id, 'name': gen_string('alpha')})
+
+        # Create repositories
+        repo_ondemand = make_repository(
+            {
+                'download-policy': 'on_demand',
+                'organization-id': module_org.id,
+                'product-id': product['id'],
+            }
+        )
+        repo_immediate = make_repository(
+            {
+                'download-policy': 'immediate',
+                'organization-id': module_org.id,
+                'product-id': product['id'],
+            }
+        )
+
+        # Sync repositories
+        Repository.synchronize({'id': repo_ondemand['id']})
+        Repository.synchronize({'id': repo_immediate['id']})
+
+        # Create cv and publish
+        cv = make_content_view({'name': gen_string('alpha'), 'organization-id': module_org.id})
+        ContentView.add_repository(
+            {
+                'id': cv['id'],
+                'organization-id': module_org.id,
+                'repository-id': repo_ondemand['id'],
+            }
+        )
+        ContentView.add_repository(
+            {
+                'id': cv['id'],
+                'organization-id': module_org.id,
+                'repository-id': repo_immediate['id'],
+            }
+        )
+        ContentView.publish({'id': cv['id']})
```

Comment:
Would suggest using a loop for ondemand/immediate policy, just to optimize the code
```suggestion
        # Create custom product, and CV
        product = make_product({'organization-id': module_org.id, 'name': gen_string('alpha')})
        cv = make_content_view({'name': gen_string('alpha'), 'organization-id': module_org.id})

        # Create repositories, sync, and add into CV
        for policy in ['on_demand', 'immediate']:
            repo = make_repository(
                {
                    'download-policy': policy,
                    'organization-id': module_org.id,
                    'product-id': product['id'],
                }
            )
            Repository.synchronize({'id': repo['id']})
            ContentView.add_repository(
                {
                    'id': cv['id'],
                    'organization-id': module_org.id,
                    'repository-id': repo['id'],
                }
            )
        ContentView.publish({'id': cv['id']})
```

Reply:
This looks nice but it removed _repo_ondemand_ and _repo_immediate_ variables which are used later during assertions. This might be solved by few more lines of code, maybe by using a dict... But I don't see more benefits that are worth the time.

Reply:
@rmynar ah, I didn't see those asserts earlier for this test, but those are not available in below test, where this can be applied, and i think this can be handled separately as its not blocking